### PR TITLE
test(mbb): pin the Torvik join the R golden's outage hid

### PR DESCRIPTION
**Pull request by saiemgilani**
## The problem

`mbb_team_crosswalk`'s parity gate could not tell a working Torvik join from a broken one.

The 2026 MBB team golden (`tests/fixtures/crosswalk_basketball/mbb_team_crosswalk_2026.parquet`) has **`bart_team` non-null on 0 of 362 rows**, and its `match_method` reads `fox+kp` (359) / `fox_only` (2) / `espn_only` (1). That is not intended behaviour: the R builder wraps its Torvik fetch in `tryCatch(..., error = ~ NULL)`, and that fetch failed the day the fixture was frozen. **The golden records a transient upstream outage.**

Python is more correct than its oracle here — `mbb_team_crosswalk()` has no such swallow (a failed Torvik fetch raises `CrosswalkSourceError`), so it joins Torvik and reports `fox+bart+kp` for the same 359 rows.

The gate did not notice, because every parity test reconstructs the provider inputs *out of the golden* — and `distinct_where(want, "bart_team", ...)` on an all-null column yields an empty frame. Both existing MBB tests therefore fed the assembler an empty Torvik input and passed. A regression back to all-null `bart_*` would have been silent.

## The fix

* Commit a real Torvik team/conf capture (`mbb_torvik_teams_2026.parquet`, 365 rows, `torvik_ratings(year=2026)`, captured 2026-08-12).
* Add `test_mbb_team_crosswalk_joins_torvik_where_the_golden_froze_an_outage`, which feeds that capture plus the bundled KenPom directory to `_assemble_team_crosswalk` and pins **both sides** of the divergence:
  * the golden's defect (`bart_team` non-null == 0, `fox+kp` == 359) — so a re-capture that fixes it fails loudly rather than drifting;
  * Python's corrected output: `bart_team` / `bart_conf` / `bart_match_confidence` non-null == 359, `match_method == "fox+bart+kp"` on 359, `fox+kp` == 0, and the three ESPN teams Torvik's directory genuinely does not carry named explicitly (LSU New Orleans, St. Thomas, West Florida);
  * row-level equality with the golden on every non-diverging column (column order and dtypes included), plus `match_method` agreeing exactly once `+bart` is stripped back out.
* Record the provenance in `tests/fixtures/crosswalk_basketball/README.md`: what the defect is, that it is a frozen outage rather than Torvik coverage, that Python deliberately diverges, that dropping the column from comparison is not an acceptable alternative, the capture date, and a re-capture recipe.
* Cross-reference from `test_mbb_team_crosswalk_bundled_kenpom_reproduces_the_golden`, which passes Torvik in empty **on purpose** and is not the authority on `bart_*`.

`BART_JOINED = 359` is measured against the committed capture, not chosen.

## Verification that the new assertion bites

Mutated `_assemble_team_crosswalk` to force the empty-Torvik branch (the exact regression the R `tryCatch` produces):

```diff
-    if bart is None or bart.height == 0:
+    if True:  # MUTATION: simulate the R tryCatch swallowing a Torvik outage
         bart2 = pl.DataFrame(schema={"bart_team": pl.Utf8, "bart_conf": pl.Utf8, "ct_key_bart": pl.Utf8})
```

```
>       assert got["bart_team"].is_not_null().sum() == BART_JOINED
E       AssertionError: assert 0 == 359

FAILED tests/test_basketball_crosswalk_parity.py::test_mbb_team_crosswalk_joins_torvik_where_the_golden_froze_an_outage
1 failed, 21 passed in 4.62s
```

Only the new test fails — confirming the other 21, including both existing MBB team-crosswalk tests, were blind to this regression. Mutation reverted.

## Checks

* `uv run pytest tests/test_basketball_crosswalk_parity.py tests/test_crosswalk_basketball_sources.py` — 76 passed
* `uv run ruff check` + `ruff format --check` — clean
* `uv run python tools/codegen/generate.py --check` — exit 0 (no codegen sources touched)
* pre-commit: all hooks passed, doctoc clean

## Summary by Sourcery

Strengthen MBB team crosswalk parity around Torvik data by pinning a known defect in the golden and asserting Python’s corrected behaviour using a committed Torvik capture.

Documentation:
- Document the known defect in the 2026 MBB team golden’s bart columns, the intended Python divergence, the role of the Torvik capture, and provide a recipe for re-capturing it in the fixtures README.

Tests:
- Add a dedicated test ensuring the MBB team crosswalk correctly joins Torvik data despite the golden having all-null bart columns, including row- and match_method-level assertions.
- Update the existing MBB team crosswalk parity test to explicitly document that it reproduces the defective golden and is not authoritative for bart-related behaviour.
- Introduce a constant for the expected count of Torvik-joined teams based on the committed capture.

## Summary by CodeRabbit

* **Tests**
  * Added regression coverage for basketball team crosswalk integration with Torvik data.
  * Verified Torvik fields, match methods, expected unmatched teams, and parity across remaining fields.
  * Added validation for the expected number of successful Torvik matches.

* **Documentation**
  * Documented the temporary missing Torvik data and offline test fixture.
  * Added guidance for regenerating the fixture, updating its capture date, and verifying results.

### CODE REVIEW

File: tests/fixtures/crosswalk_basketball/README.md [hunk 1]
```diff
@@ -38,8 +39,55 @@ Note the `<lg>/crosswalk/*.csv` files in those repos are per-season **manifests*
...
 The parquets above are the actual outputs.
 
+## Known defect in the MBB team golden: all-null bart columns
+
+Every one of its 362 rows has `bart_team` / `bart_conf` /
+`bart_match_confidence` **null**, and `match_method` reads `fox+kp` (359) /
+`fox_only` (2) / `espn_only` (1). That is **not** intended behaviour and not a
+statement about Torvik coverage — the R builder wraps its Torvik fetch in
+`tryCatch(..., error = ~ NULL)`, and that fetch failed on the day the R
+producer ran. The golden froze a transient upstream outage.
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply tests/fixtures/crosswalk_basketball/README.md:50]: Valid — fixed in 5cc7b60.

Checked against the actual R source rather than the paraphrase. `hoopR/R/mbb_crosswalk.R:346-347` reads:

```r
bart_raw <- tryCatch(torvik_ratings(year = season),
                     error = function(e) NULL)
```

So the handler was always callable; `error = ~ NULL` was my inaccurate shorthand in the prose, not something the producer runs. Left uncorrected it also misleads on the mechanism — a formula handler would not swallow the outage at all, it would raise `attempt to apply non-function`, which is the opposite of the behaviour the fixture froze.

Corrected at all three sites (the README had one, the test file two — the second read `tryCatch(..., = NULL)`), and each now quotes the call verbatim with a `hoopR/R/mbb_crosswalk.R:346-347` citation so the next reader can check it against the source instead of trusting the paraphrase.

File: tests/fixtures/crosswalk_basketball/README.md [hunk 2]
```diff
@@ -38,8 +39,55 @@ Note the `<lg>/crosswalk/*.csv` files in those repos are per-season **manifests*
...
+
+sdv-py deliberately **diverges**: `mbb_team_crosswalk()` has no such swallow (a
+failed Torvik fetch raises `CrosswalkSourceError`), so it joins Torvik and
+reports `fox+bart+kp` for the same 359 rows. Do not "fix" Python to reproduce
+the nulls, and do not drop `bart_*` from the comparison — dropping it silently
+permits a regression back to all-null.
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply tests/fixtures/crosswalk_basketball/README.md]: Valid — the contradiction was real, fixed in 5cc7b60.

The README sentence and the test disagreed, and the README was the wrong one. Dropping `bart_*` and `match_method` at L299-300 is correct: that diff is an exactness comparison against a golden that is *known to be wrong* in exactly those columns, so including them would assert the defect.

Rewrote it to state the actual rule rather than a blanket prohibition — the bart columns must be **positively asserted** (populated counts, `match_method` composition), which the test does at L285-295 before the diff; excluding them from the diff afterwards is fine. What is not acceptable is dropping them and asserting nothing, which is the regression-to-all-null hazard the original sentence was reaching for.

Took the substance of your wording rather than the literal diff, since "assert the corrected fields first, then compare the remaining columns" is the rule and I wanted the README to say *why* the exclusion is legitimate.

File: tests/fixtures/crosswalk_basketball/README.md [hunk 3]
```diff
@@ -38,8 +39,55 @@ Note the `<lg>/crosswalk/*.csv` files in those repos are per-season **manifests*
...
+
+`mbb_torvik_teams_2026.parquet` is not an R output — re-capture it with:
+
+```python
+import polars as pl
+from sportsdataverse.mbb.torvik import torvik_ratings
+
+torvik_ratings(year=2026).select(
+    pl.col("team").cast(pl.Utf8), pl.col("conf").cast(pl.Utf8)
+).write_parquet("mbb_torvik_teams_2026.parquet")
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply tests/fixtures/crosswalk_basketball/README.md]: Valid — fixed in 5cc7b60.

Applied the repository-relative path and stated the run-from-root precondition explicitly:

```python
).write_parquet("tests/fixtures/crosswalk_basketball/mbb_torvik_teams_2026.parquet")
```

The failure mode is worse than a stray file: the re-capture appears to succeed, the committed fixture stays stale, and `BART_JOINED` then gets re-measured against a capture that was never actually refreshed — silently pinning the floor to old data.

File: tests/test_basketball_crosswalk_parity.py [hunk 4]
```diff
@@ -229,6 +242,68 @@ def test_mbb_team_crosswalk_bundled_kenpom_reproduces_the_golden() -> None:
...
+    fixture README) and pins the divergence in both directions: the golden's
+    defect and Python's corrected output.
+    """
+    want = golden("mbb_team")
+    bart = pl.read_parquet(FIXTURES / f"mbb_torvik_teams_{SEASON}.parquet")
+
+    # The golden's recorded defect. If a re-capture ever fixes it, this fails
+    # and the whole test should collapse back into the plain parity gate.
+    assert want["bart_team"].is_not_null().sum() == 0
+    assert want.filter(pl.col("match_method") == "fox+kp").height == BART_JOINED
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply tests/test_basketball_crosswalk_parity.py]: Valid — fixed in 5cc7b60, with a mutation proof.

The baseline now asserts all three frozen Torvik columns:

```python
for column in ("bart_team", "bart_conf", "bart_match_confidence"):
    assert want[column].is_null().all(), f"{column} is not all-null in the frozen golden"
```

Since this is a test-hardening change, I proved it rather than asserting it works. Built a golden variant where `bart_team` stays all-null (so the defect still *looks* intact) but `bart_conf` is populated on 5 rows, pointed the test's `FIXTURES` at it, and ran the test both ways:

```
# before (assert want["bart_team"].is_not_null().sum() == 0)
RESULT: ACCEPTED (partially-populated golden passed the baseline check)

# after
RESULT: REJECTED (test failed as it should) -- bart_conf is not all-null in the frozen golden
```

So the old check demonstrably admitted a golden that is no longer a clean record of the outage, and the new one rejects it. Full suite still green: 22/22 in `tests/test_basketball_crosswalk_parity.py`.
